# Gains associated with the low-level controller
# For the reaction wheeels multiplying orientation error and angular rate
Orientation:
  Kp: [ -90,-90,-90 ]
  Kd: [ -6,-6,-6]
# For the leg multiplying deflection error and velocity
Leg:
  Kp: .4
  Kd: 0.02
LowLevel:
  dt: 0.001

Simulator:
  p0: [-3, -5, 0.5]         # initial position [x,y,z]
  v0: [0,0,0]           # initial velocity [x_dot, y_dot, z_dot]
  rpy0: [0,0,0]         # initial orientation [roll, pitch, yaw]
  w0: [0,0,0]           # initial angular rate [roll_dot, pitch_dot, yaw_dot]
  pauseBeforeStart: 0   # how long to wait [in seconds] before simulation begins
  speed: 1              # speed of simulation [multiplier]

  # Perturbation Forces
  pert_force_x: [0, 0, 0, 0]    # perturbation force in X direction at below times
  pert_force_y: [0, 0, 0, 0]    # perturbation force in Y direction at below times
  pert_start: [2000, 5000, 11000,14000] # start time for force [milliseconds]
  pert_end: [2700, 5700, 11150, 14150]  # end time for force [milliseconds
  simEnd: 10000.0               # time to end the simulation [s]
  yaw_drift: 0     # drift in rad / s

RaibertHeuristic:
  # kx_p: -0.2                                            # p gain for placing foot to manipulate the forward velocity (x-direction)
  # ky_p: 0.2                                             # p gain for placing foot to manipulate the forward velocity (y-direction)
  # kx_d: -0.4                                            # d gain for placing foot to manipulate the forward velocity (x-direction)
  # ky_d: 0.4                                             # d gain for placing foot to manipulate the forward velocity (y-direction)
  # kx_f: 0.5                # feed forward gain for velocity tracking (x-direction) as a scaling of ky_d
  # ky_f: 0.5              # feed forward gain for velocity tracking (y-direction) as a scaling of ky_d
  kx_p: -0.1                                            # p gain for placing foot to manipulate the forward velocity (x-direction)
  ky_p: 0.1                                             # p gain for placing foot to manipulate the forward velocity (y-direction)
  kx_d: -0.3                                            # d gain for placing foot to manipulate the forward velocity (x-direction)
  ky_d: 0.3                                             # d gain for placing foot to manipulate the forward velocity (y-direction)
  kx_f: 0.5                # feed forward gain for velocity tracking (x-direction) as a scaling of ky_d
  ky_f: 0.5              # feed forward gain for velocity tracking (y-direction) as a scaling of ky_d
  pos_clip: 0.5
  vel_clip: 1
  v_des_clip: 0.3
  angle_max: 0.6
  yaw_damping: 0.0035

PredCBF:
  horizon: 5
  dt: 0.01
  alpha: 10
  rho: 1
  smooth_barrier: False
  epsilon: 0
  iters: 1
  K: 0.5
  tol: 0.001
  k_r: 1
  v_max: 0.3
  # v_max: 10
  pred_dt: 0.3
  rs: [0.5, 0.5]
  cxs: [-0.5, -2]
  cys: [-2, -3]
  zd: [0, 0]
  use_delta: True
  use_barrier: True

SpringStiffness: 1000 # Not so unreasonable that this is a parameter, as it is a soft contact simulator

roll_offset: 0.00
pitch_offset: -0.00

Policy:
  dt_policy: 0.02

dt_planner: 0.2